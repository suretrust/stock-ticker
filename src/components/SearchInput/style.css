input {
    width: 382px;
    max-width: 440px;
    min-width: 220px;
    height: 54px;
    background: #F5F8FA;
    border-radius: 8px;
    border: none;
    background-image: url('../../images/search-icon.svg');
    background-repeat: no-repeat;
    background-position: 17.47px;
    padding-left: 56px;
    font-size: 15px;
    line-height: 24px;
    color: #00264C;
    margin-bottom: 8px;
}

input:focus {
    outline: none;
}

input::placeholder {
    font-size: 13px;
    line-height: 16px;
    color: #728BA3;
    opacity: 1;
}