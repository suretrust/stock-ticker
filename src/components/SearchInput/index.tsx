import React from 'react';
import './style.css';

interface SearchInputProps {
  searchValue: string;
  handleSearchChange: React.ChangeEventHandler<HTMLInputElement>;
}

const SearchInput: React.FC<SearchInputProps> = ({
  searchValue,
  handleSearchChange,
}) => {
  return (
    <input
      autoFocus
      value={searchValue}
      onChange={handleSearchChange}
      type='text'
      placeholder='Search for stock ticker'
    />
  );
};

export default SearchInput;
