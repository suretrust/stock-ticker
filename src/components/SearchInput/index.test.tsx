import { render, screen } from '@testing-library/react';
import SearchInput from '.';

describe('SearchInput component', () => {
  it('renders SearchInput as fragment', () => {
    const { asFragment } = render(
      <SearchInput searchValue='test' handleSearchChange={() => jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders input element placeholder text', () => {
    render(
      <SearchInput searchValue='test' handleSearchChange={() => jest.fn()} />
    );
    const inputElement = screen.getByPlaceholderText(
      /Search for stock ticker/i
    );

    expect(inputElement).toBeTruthy();
  });
});
