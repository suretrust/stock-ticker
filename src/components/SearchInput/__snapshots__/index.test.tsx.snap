// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component renders SearchInput as fragment 1`] = `
<DocumentFragment>
  <input
    placeholder="Search for stock ticker"
    type="text"
    value="test"
  />
</DocumentFragment>
`;
